torch/distributed/_spmd/graph_optimization.py

```diff
@@ -411,6 +412,19 @@ def _fuse_with_cat(
     return fused_comm_block
 
 
+def _expedite_comm_ops(gm: IterGraphModule, comm_blocks: List[CommBlock]) -> None:
+    node_indices = {node: i for i, node in enumerate(gm.graph.nodes)}
+    for comm_block in comm_blocks:
+        last_input = comm_block.comm_node
+        last_input_idx = -1
+        for input in comm_block.inputs:
+            input_idx = node_indices[input]
+            if input_idx > last_input_idx:
+                last_input = input
+                last_input_idx = input_idx
+        gm.graph.node_append(last_input, comm_block.comm_node)
```

Comment:
`node_append` is in `IterGraph`. Right now, `IterGraph` overwrites all the methods of `fx.Graph` but `append` is a node level method. The current solution is to let `IterGraph` provides node level methods by prefixing them with `node_`, hence `node_append`. This is annoying. So I plan to create a `IterNode` to overwrite all the methods so that optimization developers do not have to use proprietary methods. 

And no, `append` does not need to erase the original node. It simply changes the node order in the `fx.Graph`.

Reply:
Oh I see